### ISSUE TITLE
Train ViT-UNet on real S2 data

### ISSUE THREAD

**Issue by ltrgoddard** [data-desk-eco/burnoff#48]
## Context

After #47 downloads real S2 patches and #46 validates the pipeline on synthetic data, the model needs to be trained on real imagery for production quality. This is the key step for achieving the paper's detection performance.

## Task

Train the ViT-UNet on real S2 patches with synthetic plume embedding.

### Steps

1. Ensure `data/tiles/` has training patches from #47
2. Run training on cloud GPU:
   ```bash
   uv run python train.py --epochs 10 --batch-size 64
   ```
   (omit `--synthetic` to use real S2 data via `S2PatchDataset`)
3. Monitor metrics every epoch
4. Export best checkpoint to ONNX:
   ```bash
   uv run python export_onnx.py checkpoints/best.pt --fp16
   ```
5. Validate ONNX output matches PyTorch (script does this automatically)

### Target metrics (from paper)

- F1 > 0.8 on test set
- Pixel-wise FPR < 0.1% (paper achieves <0.03%)
- Detects plumes down to ~0.01 km² area

### Notes

- Paper: 1.235M samples, batch 64, 10 epochs, Adam lr=1e-3
- LR reduces by 0.1% after 10 stale batches (configured in scheduler)
- `S2PatchDataset` randomly pairs patches from different dates and embeds plumes in 50%
- The dataset returns `(input_tensor, target_mask, metadata)` — the training loop already handles the 2-tuple from synthetic and 3-tuple from real data

### Depends on

- #46 (synthetic training validates pipeline)
- #47 (real S2 data downloaded)

## Branch

`methane-vit-unet`
